pkg/pgmodel/cache/inverted_labels_cache.go

```diff
@@ -0,0 +1,63 @@
+package cache
+
+import (
+	"fmt"
+
+	"github.com/dgraph-io/ristretto"
+)
+
+type LabelInfo struct {
+	LabelID int32
+	Pos     int32
+}
+
+type LabelKey struct {
+	MetricName, Name, Value string
+}
+
+func NewLabelKey(metricName, name, value string) LabelKey {
+	return LabelKey{MetricName: metricName, Name: name, Value: value}
+}
+
+// Cache hash function needs string to work with
+func (lk LabelKey) String() string {
+	return fmt.Sprintf("%s#%s=%s", lk.MetricName, lk.Name, lk.Value)
+}
+
+func NewLabelInfo(lableID, pos int32) LabelInfo {
+	return LabelInfo{LabelID: lableID, Pos: pos}
+}
+
+// Label key-pair -> (id,pos) cache
+// Used when creating series to avoid DB calls for labels
+type InvertedLabelsCache struct {
+	cache *ristretto.Cache
+}
+
+// Cache is thread-safe
+func NewInvertedLablesCache(size int) (*InvertedLabelsCache, error) {
+	if size == 0 {
+		return nil, fmt.Errorf("labels cache size must be > 0")
+	}
+	cache, _ := ristretto.NewCache(&ristretto.Config{ // TODO: pass cache config
```

Comment:
If we are returning error, its good to check the error here too. This will protect from future implementation changes in the new cache lib.